### PR TITLE
Upgrade to target compile 17

### PR DESCRIPTION
**Pull request by fsamuel-bs**
## General
**After this PR**:
Upgrade AtlasDB to java 17. Follow up of https://github.com/palantir/atlasdb/pull/7306.

Changes:
- remove `com.google.common.primitives.Longs` references
- upgrade several dependencies

What needs to be done in the future:
- upgrade to Dagger 2.52+ done in conjunction to adding jakarta dependencies. Because `dagger.internal.Provider` added a dependency on `jakarta.inject.Inject`.
- upgrade to junit 5.11.0+ needs to be accompanied with a dropwizard upgrade to 3.0+. Because `DropwizardExtensionsSupport` uses an internal method from junit that has been removed.
    -  note that `org.mockito:* = 5.14.1` also bumps junit to 5.11.0+

==COMMIT_MSG==
==COMMIT_MSG==

### CODE REVIEW

File: versions.props [hunk 1]
```diff
@@ -102,9 +112,3 @@ org.postgresql:postgresql = 42.7.3
 
 # transitive dep with API breaks in 0.14.0
 org.apache.thrift:libthrift = 0.17.0
-
-# Large internal product uses 2.x
-org.codehaus.groovy:* = 2.5.14
```
Comment (fsamuel-bs) [inline versions.props:107]: Confirmed they have upgraded to 3.x

File: versions.props [hunk 2]
```diff
@@ -102,9 +112,3 @@ org.postgresql:postgresql = 42.7.3
...
-
-# Subsequent versions use Java 17, which we don't use right now.
-com.palantir.goethe:* = 0.12.0
```
Comment (fsamuel-bs) [inline versions.props:110]: We're now in java 17

File: versions.props [hunk 3]
```diff
@@ -94,6 +91,19 @@ io.dropwizard:* = 2.1.7
...
+org.junit.jupiter:* = 5.10.2
+org.junit.platform:* = 1.10.2
+org.mockito:* = 5.12.0
+
+# Dagger 2.52.0+ adds a dependency on jakarta. Bump this with the javax -> jakarta migration.
+com.google.dagger:* = 2.51.1
+
+# Large internal consumer still relies on protobuf 3.x.
+# We need to bump this in coordination with them.
+com.google.protobuf:* = 3.25.4
```
Comment (fsamuel-bs) [inline versions.props:105]: Pinned from above to 3.x

File: versions.props [hunk 4]
```diff
@@ -94,6 +91,19 @@ io.dropwizard:* = 2.1.7
...
 ch.qos.logback:* = 1.2.11
 
+# Dropwizard 2.1.x DropwizardExtensionsSupport relies on junit 5.10.x and below
+# All the below bump junit to an incompatible version. Bump this after bumping Dropwizard to 3.x
+org.junit.jupiter:* = 5.10.2
+org.junit.platform:* = 1.10.2
+org.mockito:* = 5.12.0
+
+# Dagger 2.52.0+ adds a dependency on jakarta. Bump this with the javax -> jakarta migration.
+com.google.dagger:* = 2.51.1
```
Comment (fsamuel-bs) [inline versions.props:101]: Explanation inline, pinned from above.

File: versions.props [hunk 5]
```diff
@@ -94,6 +91,19 @@ io.dropwizard:* = 2.1.7
 # Thus we cannot upgrade logback until dropwizard supports jakarta servlet 5.x.x
 ch.qos.logback:* = 1.2.11
 
+# Dropwizard 2.1.x DropwizardExtensionsSupport relies on junit 5.10.x and below
+# All the below bump junit to an incompatible version. Bump this after bumping Dropwizard to 3.x
+org.junit.jupiter:* = 5.10.2
+org.junit.platform:* = 1.10.2
+org.mockito:* = 5.12.0
```
Comment (fsamuel-bs) [inline versions.props:98]: Explanation inline, pinned from above.

File: versions.props [hunk 6]
```diff
@@ -10,73 +10,70 @@ com.google.auto.service:auto-service-annotations = 1.0
...
+one.util:streamex = 0.8.3
+org.apache.commons:commons-lang3 = 3.17.0
 org.apache.commons:commons-pool2 = 2.12.0
 org.apache.commons:commons-text = 1.12.0
 org.apache.httpcomponents.client5:httpclient5 = 5.3.1
-org.assertj:* = 3.26.0
-org.awaitility:awaitility = 4.2.1
+org.assertj:* = 3.26.3
+org.awaitility:awaitility = 4.2.2
+org.codehaus.groovy:* = 3.0.9
```
Comment (fsamuel-bs) [inline versions.props:57]: Unpinned from below.

File: versions.props [hunk 7]
```diff
@@ -10,73 +10,70 @@ com.google.auto.service:auto-service-annotations = 1.0
...
 com.palantir.config.crypto:* = 6.0.0
 com.palantir.conjure.java.api:* = 2.53.0
-com.palantir.conjure.java.runtime:* = 8.12.0
+com.palantir.conjure.java.runtime:* = 8.15.0
 com.palantir.conjure.java:* = 8.28.0
 com.palantir.conjure:conjure = 4.49.0
-com.palantir.dialogue:* = 3.135.0
+com.palantir.dialogue:* = 4.1.0
 com.palantir.docker.proxy:* = 1.6.0
+com.palantir.goethe:* = 0.12.0
```
Comment (fsamuel-bs) [inline versions.props:27]: Unpinned from below.

File: versions.props [hunk 8]
```diff
@@ -10,73 +10,70 @@ com.google.auto.service:auto-service-annotations = 1.0
...
 com.lmax:disruptor = 3.4.2
 com.palantir.common:streams = 2.4.0
 com.palantir.config.crypto:* = 6.0.0
 com.palantir.conjure.java.api:* = 2.53.0
-com.palantir.conjure.java.runtime:* = 8.12.0
+com.palantir.conjure.java.runtime:* = 8.15.0
 com.palantir.conjure.java:* = 8.28.0
 com.palantir.conjure:conjure = 4.49.0
-com.palantir.dialogue:* = 3.135.0
+com.palantir.dialogue:* = 4.1.0
```
Comment (fsamuel-bs) [inline versions.props:25]: Major version bump, but it's fine since most internal products are on 4.x already.

File: versions.props [hunk 9]
```diff
@@ -10,73 +10,70 @@ com.google.auto.service:auto-service-annotations = 1.0
 com.google.auto:auto-common = 1.2.2
 com.google.code.findbugs:findbugs-annotations = 3.0.1
 com.google.code.findbugs:jsr305 = 3.0.2
-com.google.dagger:* = 2.51.1
 # Ksp version is temporary, can be removed once Maven calms down.
 com.google.devtools.ksp:symbol-processing-api = 1.5.21-1.0.0-beta07
 com.google.errorprone:* = 2.10.0 # upgrading to 2.11.0 is blocked by a google bug
 com.google.errorprone:error_prone_annotations = 2.10.0
-com.google.guava:guava = 33.2.1-jre
-com.google.protobuf:* = 3.25.3
```
Comment (fsamuel-bs) [inline versions.props:19]: Pinned below.

File: versions.props [hunk 10]
```diff
@@ -10,73 +10,70 @@ com.google.auto.service:auto-service-annotations = 1.0
 com.google.auto:auto-common = 1.2.2
 com.google.code.findbugs:findbugs-annotations = 3.0.1
 com.google.code.findbugs:jsr305 = 3.0.2
-com.google.dagger:* = 2.51.1
```
Comment (fsamuel-bs) [inline versions.props:13]: Pinned below.

File: atlasdb-processors/src/main/java/com/palantir/processors/AutoDelegateProcessor.java [hunk 11]
```diff
@@ -96,7 +96,7 @@ public Set<String> getSupportedAnnotationTypes() {
 
     @Override
     public SourceVersion getSupportedSourceVersion() {
-        return SourceVersion.RELEASE_11;
+        return SourceVersion.RELEASE_17;
```
Comment (carterkozak) [inline atlasdb-processors/src/main/java/com/palantir/processors/AutoDelegateProcessor.java]: we probably don't want to bump this every time we upgrade the source compat target
```suggestion
        return SourceVersion.latestSupported();
```

File: versions.props (hunk 8 above)
Comment (jeremyk-91) [reply versions.props:25]: yeah confirmed large product is on 4.x

File: versions.lock [hunk 12]
```diff
@@ -61,64 +61,63 @@ com.palantir.cassandra:cassandra-thrift:2.2.18-1.108.0 (2 constraints: 0f180ff6)
...
+com.palantir.tritium:tritium-slf4j:0.93.0 (2 constraints: 7e20193e)
+com.palantir.tritium:tritium-time:0.93.0 (1 constraints: 840e8c59)
+com.palantir.tritium:tritium-tracing:0.93.0 (2 constraints: 7e20193e)
 com.squareup:javapoet:1.13.0 (4 constraints: 102cc1ed)
 com.squareup:kotlinpoet:1.11.0 (1 constraints: 790e1a5d)
 com.thinkaurelius.thrift:thrift-server:0.3.7 (1 constraints: 980f6c88)
-com.zaxxer:HikariCP:5.1.0 (1 constraints: 08050736)
-commons-cli:commons-cli:1.8.0 (3 constraints: 7e249d97)
-commons-codec:commons-codec:1.17.0 (2 constraints: 6d148085)
+com.zaxxer:HikariCP:6.0.0 (1 constraints: 08050936)
```
Comment (jeremyk-91) [inline versions.lock]: 👀 Do we know what changed here?

**Review by jeremyk-91 — APPROVED** [review]
Thanks! Couple of small bits 👍

File: versions.lock (hunk 12 above)
Comment (ergo14) [reply versions.lock]: Tagging @berler @CVdV-au in case they can advise

Comment (ergo14) [reply versions.lock]: [riCP/blob/dev/CHANGES](https://github.com/brettwooldridge/HikariCP/blob/dev/CHANGES)

**Review by ergo14 — COMMENTED** [review]
thanks for the sign posted bumps, looks good to me modulo the hikari bump

**Pull request merged**